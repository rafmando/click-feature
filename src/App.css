* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: 'Roboto', sans-serif;
}

.container {
  margin-top: 50px;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

button {
  margin-bottom: 30px;
  padding: 10px;
  border: 1px solid #ff4c5c;
  background: none;
  color: #ff4c5c;
  cursor: pointer;
}

li {
  display: flex;
  width: 300px;
  justify-content: space-between;
  padding: 10px;
  align-items: center;
  background-color:#ff4c5c;
  color: white;
  margin-bottom: 5px;
  
}

@media screen and (max-width: 350px) {
  li {
    width: 200px;
  }
}