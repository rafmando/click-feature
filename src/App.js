import './App.css';
import { useState } from 'react'

const data = [
  { name: 'Item A', price: 125 },
  { name: 'Item B', price: 230 },
  { name: 'Item C', price: 295 },
  { name: 'Item D', price: 245 },
  { name: 'Item E', price: 900 },
  { name: 'Item F', price: 875 },
  { name: 'Item G', price: 235 },
  { name: 'Item H', price: 400 },
];

function App() {

  const lowToHighProds = data.sort((a,b) => {
    if(a.price > b.price){
      return 1
    } else {
      return -1
    }
  }).map((prod) => {
    return prod
  })

  const firstFiveProds = lowToHighProds.slice(0,5)
  const [products,setProducts] = useState(firstFiveProds)
  const [buttonText,setButtonText] = useState('View All')
  const newButtonText = buttonText === 'View All' ? 'Collapse' : 'View All'

  const handleClick = () => {
    setButtonText(newButtonText)

    if(buttonText === 'Collapse') {
      setProducts(firstFiveProds)
    } else if(buttonText === 'View All') {
      setProducts(lowToHighProds)
    }
    
  }

  return (
   <div className="container">
   <button onClick={handleClick}>{buttonText}</button>
   <ul aria-label="products-list">
     {products.map((prod) => {
       return (
         <li key={prod.name}>
           <p>{prod.name}</p>
           <p>Â£{prod.price}</p>
         </li>
       )
     })}
   </ul>
   </div>
  );
}

export default App;
