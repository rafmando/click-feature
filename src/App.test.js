import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';

describe('Button unclicked', () => {
  it('renders button correctly', () => {
    render(<App />);

    const button = screen.getByRole('button', {name: 'View All'})
    expect(button.textContent).toBe('View All')
  });
  
  it('renders data correctly', () => {
    render(<App />)

    const productList = screen.getByRole('list', {
      name: /products/i,
    })
  
    expect(productList).toMatchSnapshot()
  })
})

describe('Button clicked', () => {
  it('renders button correctly when clicked', () => {
    render(<App/>);

    const button = screen.getByRole('button', {name: 'View All'})
    expect(button.textContent).toBe('View All')

    fireEvent.click(button)
    expect(button.textContent).toBe('Collapse')
  });
  
  it('renders data correctly when clicked', () => {
    render(<App />)

    const productList = screen.getByRole('list', {
      name: /products/i,
    })

    const button = screen.getByRole('button', {name: 'View All'})
    fireEvent.click(button)
    expect(productList).toMatchSnapshot()
  })
})




