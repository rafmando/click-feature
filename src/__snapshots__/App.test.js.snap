// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button clicked renders data correctly when clicked 1`] = `
<ul
  aria-label="products-list"
>
  <li>
    <p>
      Item A
    </p>
    <p>
      £
      125
    </p>
  </li>
  <li>
    <p>
      Item B
    </p>
    <p>
      £
      230
    </p>
  </li>
  <li>
    <p>
      Item G
    </p>
    <p>
      £
      235
    </p>
  </li>
  <li>
    <p>
      Item D
    </p>
    <p>
      £
      245
    </p>
  </li>
  <li>
    <p>
      Item C
    </p>
    <p>
      £
      295
    </p>
  </li>
  <li>
    <p>
      Item H
    </p>
    <p>
      £
      400
    </p>
  </li>
  <li>
    <p>
      Item F
    </p>
    <p>
      £
      875
    </p>
  </li>
  <li>
    <p>
      Item E
    </p>
    <p>
      £
      900
    </p>
  </li>
</ul>
`;

exports[`Button unclicked renders data correctly 1`] = `
<ul
  aria-label="products-list"
>
  <li>
    <p>
      Item A
    </p>
    <p>
      £
      125
    </p>
  </li>
  <li>
    <p>
      Item B
    </p>
    <p>
      £
      230
    </p>
  </li>
  <li>
    <p>
      Item G
    </p>
    <p>
      £
      235
    </p>
  </li>
  <li>
    <p>
      Item D
    </p>
    <p>
      £
      245
    </p>
  </li>
  <li>
    <p>
      Item C
    </p>
    <p>
      £
      295
    </p>
  </li>
</ul>
`;
